R/compile_sim_cpp.R

```diff
@@ -253,7 +253,7 @@ compile_sim_cpp <- function(
   }
   cpp_code[idx18] <- paste0(
     "  boost::array<double, ",size,"> b = { ", paste(rep(0, size), collapse=", ")," };\n",
-    "  const state_type& A_dum = b;\n",
+    "  state_type& A_dum = b;\n",
```

Comment:
this cannot be a `const` anymore, since we're changing it to hold the initial state and not just zeroes.